### PR TITLE
Spaced repo foldername fix

### PR DESCRIPTION
**Pull request by zionyx**
* Supports git repositories with spaces.
* Added support functionalities for the tests.
* Added tests to support functionalities, and other related tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # dockerbuild-pwsh
 Powershell Core script to build and test Docker aimges.
+
+# Development environment setup
+
+* Install PowerShell Core 6.x latest
+* Run `.\Install-Prerequisites.ps1`
```
Comment (rasmusjelsgaard) [inline README.md]: 👏

File: Tests/Find-ImageName.Tests.ps1 [hunk 2]
```diff
@@ -1,13 +1,101 @@
...
+        '/tmp'
+    }
+}
+
+function New-RandomFolder {
+    param (
+        [int]$FolderLength = 8
+    )
+    do {
+        $randomString = ( -join ((0x30..0x39) + ( 0x41..0x5A) + ( 0x61..0x7A) | Get-Random -Count $FolderLength | % {[char]$_}) )
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: I would suggest using GUID to come up with a globally unique folder name instead, see https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-guid?view=powershell-6

File: Tests/Find-ImageName.Tests.ps1 [hunk 3]
```diff
@@ -1,13 +1,101 @@
...
+        $randomPath = Join-Path $(Get-TempPath) $randomString
+    } while (Test-Path -Path $randomPath -PathType Container)
+    New-Item -Path $randomPath -ItemType Directory | Out-Null
+    return $randomPath
+}
+
+function Fake-GitRepository {
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]:
```suggestion
function New-FakeGitRepository {
```

File: Tests/Find-ImageName.Tests.ps1 [hunk 4]
```diff
@@ -1,13 +1,101 @@
...
+    param (
+        [ValidateNotNullOrEmpty()]
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]:
```suggestion
        [Parameter(Mandatory=$true)]
```

File: Tests/Find-ImageName.Tests.ps1 [hunk 5]
```diff
@@ -1,13 +1,101 @@
 Import-Module -Force $PSScriptRoot/../Docker.Build.psm1
 
+function Get-TempPath {
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: I suggest moving the functions here to a file of their own. They do not belong next to the actual tests in Describe, Context and It blocks.

File: Tests/Find-ImageName.Tests.ps1 [hunk 6]
```diff
@@ -1,13 +1,101 @@
...
+
 Describe 'Parse context from git repository' {
 
+    Context 'Validating support functions for - When git is installed' {
+
+        It 'can get TEMP path' {
+            $tempPath = Get-TempPath
+
+            if ($IsWindows) {
+                $tempPath.IndexOf($Env:TEMP) | Should -Be 0
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: When reading this, I wasn't sure what it is testing. Can you think of a way to make it clearer what you are verifying?

File: Tests/Find-ImageName.Tests.ps1 [hunk 7]
```diff
@@ -1,13 +1,101 @@
...
+                $tempPath.IndexOf('/tmp') | Should -Be 0
+            }
+        }
+
+        It 'can create random folder in TEMP folder' {
+            $pathInTemp = New-RandomFolder
+
+            Test-Path -Path $pathInTemp | Should -Be $true
+
+            Remove-Item $pathInTemp -Recurse -Force | Out-Null
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: This sort of teardown code probably belongs in an AfterAll block.

File: Tests/Find-ImageName.Tests.ps1 [hunk 8]
```diff
@@ -1,13 +1,101 @@
...
+    }
+
     Context 'When git is installed' {
 
         It 'can find a repository origin' {
-            $repositoryBasePath = (Split-Path -Parent $PSScriptRoot)
-            $result = Find-ImageName -RepositoryPath $repositoryBasePath
+            $tempFolder = New-RandomFolder
+
+            $gitRepoPath1 = Join-Path $tempFolder "NoSpace"
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]:
```suggestion
            $gitRepoPathWithoutSpace = Join-Path $tempFolder "NoSpace"
```

File: Tests/Find-ImageName.Tests.ps1 [hunk 9]
```diff
@@ -1,13 +1,101 @@
...
+            $result = Find-ImageName -RepositoryPath $gitRepoPath1
+            $result | Should -BeExactly "dockerbuild-pwsh"
+
+            Remove-Item $tempFolder -Recurse -Force | Out-Null
+        }
+
+        It 'can find a repository origin with folder with space' {
+            $tempFolder = New-RandomFolder
+
+            $gitRepoPath2 = Join-Path $tempFolder "With Space"
```
Comment (rasmusjelsgaard) [inline Tests/Find-ImageName.Tests.ps1]: $gitRepoPathWithSpace

File: Tests/Find-ImageName.Tests.ps1 (hunk 6 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: To test if `[system.io.path]::GetTempPath()` == `$Env:TEMP` value on Windows, and `/tmp` on Linux.
I want to test `Get-TempFolder` function so it is able to be tested on both Windows and on Linux.

File: Tests/Find-ImageName.Tests.ps1 (hunk 7 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: I would love to, but i need to get access to the `$tempFolder` variable during build up and tear down.

Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Got it. Worked as expected.

File: Tests/Find-ImageName.Tests.ps1 (hunk 2 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Done!

File: Tests/Find-ImageName.Tests.ps1 (hunk 3 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Done.

File: Tests/Find-ImageName.Tests.ps1 (hunk 4 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Added mandatory.

File: Tests/Find-ImageName.Tests.ps1 (hunk 5 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Moved to its own file, `Utilities.ps1` + its tests.

File: Tests/Find-ImageName.Tests.ps1 (hunk 8 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Done.

File: Tests/Find-ImageName.Tests.ps1 (hunk 9 above)
Comment (zionyx) [reply Tests/Find-ImageName.Tests.ps1]: Done.

File: Private/Utilities.ps1 [hunk 10]
```diff
@@ -0,0 +1,38 @@
+#Requires -PSEdition Core -Version 6
+
+function Get-TempPath {
```
Comment (casz) [inline Private/Utilities.ps1]: nit: there is method for this. finding docs

Comment (casz) [reply Private/Utilities.ps1]: `Get-PSDrive Temp`

Though only available in PS Core 7+

Comment (zionyx) [reply Private/Utilities.ps1]: Yeah. Tried `[system.io.path]::GetTempPath()` but only works on Windows even if .Net core is installed on Linux.

**Pull request merged**